dist/main.js

```diff
@@ -7517,62 +7500,91 @@ angular.module('znk.infra-web-app.onBoarding').run(['$templateCache', function($
     'use strict';
 
     angular.module('znk.infra-web-app.purchase').service('purchaseService',
-        ["$q", "$mdDialog", "$filter", "InfraConfigSrv", "ENV", "$log", "$mdToast", "$window", "PopUpSrv", "znkAnalyticsSrv", "StorageSrv", "AuthService", function ($q, $mdDialog, $filter, InfraConfigSrv, ENV, $log, $mdToast, $window, PopUpSrv, znkAnalyticsSrv, StorageSrv, AuthService) {
+        ["$rootScope", "$state", "$q", "$mdDialog", "$filter", "InfraConfigSrv", "ENV", "$log", "$mdToast", "$window", "PopUpSrv", "znkAnalyticsSrv", "StorageSrv", "AuthService", "PurchaseStateEnum", function ($rootScope, $state, $q, $mdDialog, $filter, InfraConfigSrv, ENV, $log, $mdToast, $window,
+                  PopUpSrv, znkAnalyticsSrv, StorageSrv, AuthService, PurchaseStateEnum) {
             'ngInject';
 
+            function getPurchasePath() {
+                if (!authData) {
+                    $log.error('Invalid user');
+                    return;
+                }
+                var path = StorageSrv.variables.appUserSpacePath + '/' + 'purchase';
+                return path.replace('$$uid', '' + authData.uid);
+            }
+            function getPendingPath() {
+                if (!authData) {
+                    $log.error('Invalid user');
+                    return;
+                }
+                var path = 'pendingPurchases/' + StorageSrv.variables.uid;
+                return path.replace('$$uid', '' + authData.uid);
+            }
```

Comment:
I think you should merge these two functions to one func with path param....they are 90% similar getPurchasePath() and getPendingPath()